ledger/acctupdates.go

```diff
@@ -1197,8 +1212,13 @@ func (au *accountUpdates) lookupWithoutRewards(rnd basics.Round, addr basics.Add
 		// against the database.
 		persistedData, err = au.accountsq.lookup(addr)
 		if persistedData.round == currentDbRound {
-			au.baseAccounts.writePending(persistedData)
-			return persistedData.accountData.GetLedgerCoreAccountData(), rnd, err
+			if persistedData.rowid != 0 {
```

Comment:
this change means that it will make a two disk lookup for this account rather than just one, right ?
( i.e. there will be another lookup in loadOld... )

I wasn't able to figure out from this PR why this change is needed. Would you mind sharing your insights ?

Reply:
ah, I see, is `baseAccounts` intended to cache non-existing accounts? `makeCompactResourceDeltas` relies on them to resolve address into rowid. So lookup op on non-existing yet account X (resource Y) for a round N puts empty records into `baseAccounts`/`baseResources` as pending, then on round N+1 they gets flushed and `makeCompactResourceDeltas` uses them to find addrid. If account X (resource Y) was created in N+1, this would cause getting zero rowid/addrid from `baseAccounts`/`baseResources` since the resolution happens before updating base cashes with actual values with results from `newAccountsRound`.
So there are two solutions: either do not cache missing records (as I did since it looked logical for me) or update `makeCompactResourceDeltas` with zero checks.

Reply:
looking on `makeCompactResourceDeltas`, the method would only use the `baseAccounts` when populating the `oldResource` for the case where the entry is missing.
The `resourcesLoadOld` will be invoked later on, and would fail making the `knownAddresses`, then, will default into making the disk lookup `addrRowidStmt` needlessly.
( so far, no failure would occur. just a wasted `addrRowidStmt` query ).
Last, in `accountsNewRound`, it *attempts to* find the new address for the resource in `newAddressesRowIDs`. If it fails here, it would need to perform the disk lookup ( i.e. the `addrRowidStmt` ).

All that being said, I suspect that you'll find that the true issue is in the way we prepare the random entries we put in the database. In particular, creating a resource for an empty account could have this outcome.

Reply:
I basically agree there is no problem with `baseAccounts` in `makeCompactResourceDeltas` but with `baseResources` for the reason stated above. If there is incomplete data in `baseResources` then it is not added to missing ones and no call to `resourcesLoadOld`. 
I'm against having additional disc lookup in `accountsNewRound`. All data needs to be resolved in advance - that's the point of having `resourcesLoadOld`

Reply:
I think that it makes a perfect sense loading all in advance at `resourcesLoadOld`; however, we would need to make a differentiation at that point between accounts that are known to be empty ( and therefore, should trigger a failure ) and accounts that aren't known yet. The `baseAccounts` cache has a limited size, and it being pruned often, hence, we shouldn't rely on its content unless we have the `accountsMu` lock taken.
what about storing in the `insertMissing` also a separate bool to indicate if the entry was known to be empty or whether it's missing ? that would allow us to make the correct call when `resourcesLoadOld` gets invoked.

Reply:
removed